azurerm/internal/services/desktopvirtualization/virtual_desktop_application_group.go

```diff
@@ -0,0 +1,191 @@
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.VirtualDesktopApplicationGroupID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DevSpaceName(),
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"tags": tags.Schema(),
+
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Desktop",
+					"RemoteApp",
+				}, false),
+			},
+
+			"host_pool_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"friendly_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 64),
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 512),
+			},
+		},
+	}
+}
+
+func resourceArmVirtualDesktopApplicationGroupCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DesktopVirtualization.ApplicationGroupsClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Virtual Desktop Host Pool creation")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Virtual Desktop Host Pool %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_virtual_host_pool", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	t := d.Get("tags").(map[string]interface{})
+
+	ApplicationGroupType := d.Get("type").(string)
+	friendlyName := d.Get("friendly_name").(string)
+	description := d.Get("description").(string)
+	hostPoolID := d.Get("host_pool_id").(string)
+
+	context := desktopvirtualization.ApplicationGroup{
+		Location: &location,
+		Tags:     tags.Expand(t),
+		ApplicationGroupProperties: &desktopvirtualization.ApplicationGroupProperties{
+			ApplicationGroupType: desktopvirtualization.ApplicationGroupType(ApplicationGroupType),
+			FriendlyName:         &friendlyName,
+			Description:          &description,
+			HostPoolArmPath:      &hostPoolID,
+		},
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resourceGroup, name, context)
+	if err != nil {
```

Comment:
could we combine these lines?
```suggestion
	if _, err := client.CreateOrUpdate(ctx, resourceGroup, name, context); err != nil {
```